drivers/azure/azure.go

```diff
@@ -0,0 +1,501 @@
+
+	// Ephemeral fields
+	ctx        *azureutil.DeploymentContext
+	resolvedIP string // cache
+}
+
+// NewDriver returns a new driver instance.
+func NewDriver(hostName, storePath string) drivers.Driver {
+	// NOTE(ahmetalpbalkan): any driver initialization I do here gets lost
+	// afterwards, especially for non-Create RPC calls. Therefore I am mostly
+	// making rest of the driver stateless by just relying on the following
+	// piece of info.
+	d := &Driver{
+		BaseDriver: &drivers.BaseDriver{
+			SSHUser:     defaultSSHUser,
+			MachineName: hostName,
+			StorePath:   storePath,
+		},
+	}
+	return d
+}
+
+// GetCreateFlags returns list of create flags driver accepts.
+func (d *Driver) GetCreateFlags() []mcnflag.Flag {
+	return []mcnflag.Flag{
+		mcnflag.StringFlag{
+			Name:   flAzureEnvironment,
+			Usage:  "Azure environment (e.g. AzurePublicCloud, AzureChinaCloud)",
+			EnvVar: "AZURE_ENVIRONMENT",
+			Value:  defaultAzureEnvironment,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureSubscriptionID,
+			Usage:  "Azure Subscription ID",
+			EnvVar: "AZURE_SUBSCRIPTION_ID",
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureResourceGroup,
+			Usage:  "Azure Resource Group name (will be created if missing)",
+			EnvVar: "AZURE_RESOURCE_GROUP",
+			Value:  defaultAzureResourceGroup,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureSSHUser,
+			Usage:  "Username for SSH login",
+			EnvVar: "AZURE_SSH_USER",
+			Value:  defaultSSHUser,
+		},
+		mcnflag.IntFlag{
+			Name:   flAzureDockerPort,
+			Usage:  "Port number for Docker engine",
+			EnvVar: "AZURE_DOCKER_PORT",
+			Value:  defaultDockerPort,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureLocation,
+			Usage:  "Azure region to create the virtual machine",
+			EnvVar: "AZURE_LOCATION",
+			Value:  defaultAzureLocation,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureSize,
+			Usage:  "Size for Azure Virtual Machine",
+			EnvVar: "AZURE_SIZE",
+			Value:  defaultAzureSize,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureImage,
+			Usage:  "Azure virtual machine OS image",
+			EnvVar: "AZURE_IMAGE",
+			Value:  defaultAzureImage,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureVNet,
+			Usage:  "Azure Virtual Network name to connect the virtual machine",
+			EnvVar: "AZURE_VNET",
+			Value:  defaultAzureVNet,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureSubnet,
+			Usage:  "Azure Subnet Name to be used within the Virtual Network",
+			EnvVar: "AZURE_SUBNET",
+			Value:  defaultAzureSubnet,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureSubnetPrefix,
+			Usage:  "Private CIDR block to be used for the new subnet, should comply RFC 1918",
+			EnvVar: "AZURE_SUBNET_PREFIX",
+			Value:  defaultAzureSubnetPrefix,
+		},
+		mcnflag.StringFlag{
+			Name:   flAzureAvailabilitySet,
+			Usage:  "Azure Availability Set to place the virtual machine into",
+			EnvVar: "AZURE_AVAILABILITY_SET",
+			Value:  defaultAzureAvailabilitySet,
+		},
+		mcnflag.StringFlag{
+			Name:  flAzurePrivateIPAddr,
```

Comment:
@dgageot yes, they either cannot really be used on multiple VMs, or they should be consciously set as an argument, or they're bool/slices so it's not really worth the trouble making them env vars.